.github/scripts/CheckPullRequest.py

```diff
@@ -0,0 +1,70 @@
+#!/usr/bin/env python
+
+# Distributed under the MIT License.
+# See LICENSE.txt for details.
+
+import github
+import re
+import unittest
+
+
+class TestPullRequest(unittest.TestCase):
+    """Tests to run over pull requests before they can be merged."""
+
+    @classmethod
+    def setUpClass(cls):
+        gh = github.Github(cls.github_token)
+        gh_repo = gh.get_repo(cls.github_repository)
+        cls.pr = gh_repo.get_pull(cls.pr_id)
+
+    def test_review_checklist_is_complete(self):
+        """Check that all items in the code-review checklist are ticked off."""
+        CHECKLIST_PATTERN = r'### Code review checklist[\r\n]+(.*)[\r\n]+#'
+        match_checklist = re.search(CHECKLIST_PATTERN,
+                                    self.pr.body,
+                                    flags=re.DOTALL)
+        self.assertIsNotNone(
+            match_checklist,
+            ("Did not find the code review checklist in the PR description, "
+             f"matching pattern: {CHECKLIST_PATTERN}"))
+        checklist = match_checklist.group(1)
+        CHECK_PATTERN = r'^- \[([ xX]?)\]'
+        checks = re.findall(CHECK_PATTERN, checklist, flags=re.MULTILINE)
+        EXPECTED_NUM_CHECKS = 3
+        self.assertEqual(
+            len(checks), EXPECTED_NUM_CHECKS,
+            (f"Expected {EXPECTED_NUM_CHECKS} checks in the review checklist, "
+             f"but found {len(checks)} matching pattern: {CHECK_PATTERN}"))
+
+        def is_checked(check):
+            return check in ['x', 'X']
+
+        num_successful_checks = len(list(filter(is_checked, checks)))
+        self.assertEqual(
+            num_successful_checks, EXPECTED_NUM_CHECKS,
+            ("Not all items in the review checklist are complete. Please "
```

Comment:
[picking random line] if I understand this correctly, it adds a check (that we ideally would make as `Required`) that someone has gone and checked off all the elements in the PR review checklist? While I agree this is good in theory, I've yet to see a single PR where anything there gets checked. It's been difficult enough to get people to properly check of the type of change a PR makes, never mind anything more. I don't think we should add this check without bringing it up at the spectre call first. My worry is that this unfortunately may deter people from submitting PRs (GUIs can be a bit of a pain ;) )

Reply:
I agree, let's bring this up next Monday to hear what people think. Note that I created a new "PR automation" workflow for this test that runs on opening _and editing_ of PRs, not on pushes to the branch. So you can actually click the checkboxes above and after a few seconds the check updates, without running any of the other tests. You can test it by pushing this branch to your fork, open a PR to your fork's develop and click the checkboxes. This means the reviewers can just tick off the boxes before merging the PR (also reminds _them_ to add appropriate labels to the PR).